### PR TITLE
azure-cli: 2.44.1 -> 2.49.0

### PR DESCRIPTION
**Pull request by jfroche**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix
Comment (jfroche) [reply pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix]: Good catch, thank you

**Review by emilylange — APPROVED** [review]

File: pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix:14]:
```suggestion
  disabled = pythonOlder "3.7";
```
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/appcontainers/azure-mgmt-appcontainers/setup.py#L54

File: pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix:36]:
```suggestion
    homepage = "https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/appcontainers/azure-mgmt-appcontainers";
```

File: pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix:38]: Please do not add other people without their permission as maintainers

File: pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix:32]:
```suggestion
  pythonImportsCheck = [ "azure.mgmt.appcontainers" ];
```
we don't check dependencies

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix
Comment (jfroche) [reply pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix]: > we don't check dependencies

ok, I could do it for all the other azure packages:

<details>
  <summary>Diff</summary>
  
  ```diff
diff --git a/pkgs/development/python-modules/azure-mgmt-apimanagement/default.nix b/pkgs/development/python-modules/azure-mgmt-apimanagement/default.nix
index 7b088728b27..9b38c857b3d 100644
--- a/pkgs/development/python-modules/azure-mgmt-apimanagement/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-apimanagement/default.nix
@@ -26,7 +26,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.apimanagement" ];
+  pythonImportsCheck = [ "azure.mgmt.apimanagement" ];
 
   meta = with lib; {
     description = "Microsoft Azure API Management Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-appconfiguration/default.nix b/pkgs/development/python-modules/azure-mgmt-appconfiguration/default.nix
index 9915297eb12..05295690f86 100644
--- a/pkgs/development/python-modules/azure-mgmt-appconfiguration/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-appconfiguration/default.nix
@@ -28,7 +28,7 @@ buildPythonPackage rec {
 
   pythonNamespaces = [ "azure.mgmt" ];
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.appconfiguration" ];
+  pythonImportsCheck = [ "azure.mgmt.appconfiguration" ];
 
   meta = with lib; {
     description = "Microsoft Azure App Configuration Management Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-botservice/default.nix b/pkgs/development/python-modules/azure-mgmt-botservice/default.nix
index 6bc3b2c7c6f..5d5c5d59e8c 100644
--- a/pkgs/development/python-modules/azure-mgmt-botservice/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-botservice/default.nix
@@ -26,7 +26,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.botservice" ];
+  pythonImportsCheck = [ "azure.mgmt.botservice" ];
 
   meta = with lib; {
     description = "Microsoft Azure API Management Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-deploymentmanager/default.nix b/pkgs/development/python-modules/azure-mgmt-deploymentmanager/default.nix
index c8f15f5960c..1dc643dd675 100644
--- a/pkgs/development/python-modules/azure-mgmt-deploymentmanager/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-deploymentmanager/default.nix
@@ -26,7 +26,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.deploymentmanager" ];
+  pythonImportsCheck = [ "azure.mgmt.deploymentmanager" ];
 
   meta = with lib; {
     description = "Microsoft Azure Deployment Manager Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-hdinsight/default.nix b/pkgs/development/python-modules/azure-mgmt-hdinsight/default.nix
index a6920815f0f..09130c01e03 100644
--- a/pkgs/development/python-modules/azure-mgmt-hdinsight/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-hdinsight/default.nix
@@ -26,7 +26,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.hdinsight" ];
+  pythonImportsCheck = [ "azure.mgmt.hdinsight" ];
 
   meta = with lib; {
     description = "Microsoft Azure HDInsight Management Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-kusto/azure-mgmt-apimanagement/default.nix b/pkgs/development/python-modules/azure-mgmt-kusto/azure-mgmt-apimanagement/default.nix
index f1effb72b4c..57a20f37f2f 100644
--- a/pkgs/development/python-modules/azure-mgmt-kusto/azure-mgmt-apimanagement/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-kusto/azure-mgmt-apimanagement/default.nix
@@ -20,7 +20,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.apimanagement" ];
+  pythonImportsCheck = [ "azure.mgmt.apimanagement" ];
 
   meta = with lib; {
     description = "Microsoft Azure API Management Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-kusto/default.nix b/pkgs/development/python-modules/azure-mgmt-kusto/default.nix
index b4321d50dba..dc8f591719f 100644
--- a/pkgs/development/python-modules/azure-mgmt-kusto/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-kusto/default.nix
@@ -26,7 +26,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.kusto" ];
+  pythonImportsCheck = [ "azure.mgmt.kusto" ];
 
   meta = with lib; {
     description = "Microsoft Azure Kusto Management Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-managedservices/default.nix b/pkgs/development/python-modules/azure-mgmt-managedservices/default.nix
index 140f08e13f4..ce836d23827 100644
--- a/pkgs/development/python-modules/azure-mgmt-managedservices/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-managedservices/default.nix
@@ -26,7 +26,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.managedservices" ];
+  pythonImportsCheck = [ "azure.mgmt.managedservices" ];
 
   meta = with lib; {
     description = "Microsoft Azure Managed Services Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-privatedns/default.nix b/pkgs/development/python-modules/azure-mgmt-privatedns/default.nix
index c6ed92e0428..259054f39c3 100644
--- a/pkgs/development/python-modules/azure-mgmt-privatedns/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-privatedns/default.nix
@@ -26,7 +26,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.privatedns" ];
+  pythonImportsCheck = [ "azure.mgmt.privatedns" ];
 
   meta = with lib; {
     description = "Microsoft Azure DNS Private Zones Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-mgmt-sqlvirtualmachine/default.nix b/pkgs/development/python-modules/azure-mgmt-sqlvirtualmachine/default.nix
index 76f0d7a2b39..1026aa00f5e 100644
--- a/pkgs/development/python-modules/azure-mgmt-sqlvirtualmachine/default.nix
+++ b/pkgs/development/python-modules/azure-mgmt-sqlvirtualmachine/default.nix
@@ -26,7 +26,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.mgmt.sqlvirtualmachine" ];
+  pythonImportsCheck = [ "azure.mgmt.sqlvirtualmachine" ];
 
   meta = with lib; {
     description = "Microsoft Azure SQL Virtual Machine Management Client Library for Python";
diff --git a/pkgs/development/python-modules/azure-multiapi-storage/default.nix b/pkgs/development/python-modules/azure-multiapi-storage/default.nix
index 933d04a4785..2e693b7ae42 100644
--- a/pkgs/development/python-modules/azure-multiapi-storage/default.nix
+++ b/pkgs/development/python-modules/azure-multiapi-storage/default.nix
@@ -30,7 +30,7 @@ buildPythonPackage rec {
   # no tests included
   doCheck = false;
 
-  pythonImportsCheck = [ "azure.common" "azure.multiapi.storage" ];
+  pythonImportsCheck = [ "azure.multiapi.storage" ];
 
   meta = with lib; {
     description = "Microsoft Azure Storage Client Library for Python with multi API version support.";
  
  ```
</details>

Comment (SuperSandro2000) [reply pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix]: Do we check if the namespacing works? That could be an exception I missed.

Comment (jfroche) [reply pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix]: I guess that was the goal, `azure.common` is just one of the dependency.
@SuperSandro2000 what should we do to unblock this ?

File: pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/azure-mgmt-appcontainers/default.nix:32]:
```suggestion
  pythonImportsCheck = [ "azure.common" "azure.mgmt.appcontainers" ];
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
fails to build:

```
ERROR: Could not find a version that satisfies the requirement azure-mgmt-resource==22.0.0 (from azure-cli) (from versions: none)
ERROR: No matching distribution found for azure-mgmt-resource==22.0.0
```

**Comment by IndeedNotJames** [comment]
That's due to https://github.com/NixOS/nixpkgs/commit/35b5ea83e3005cfc58ae9b3e545195608ea276c8 (#226675)

Also, https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.48.1 has been released in the meantime.
From what I can tell, it still wants `azure-mgmt-resource==22.0.0`[^1][^2].

But you can try adding another `--replace` to
https://github.com/NixOS/nixpkgs/blob/1a411f23ba299db155a5b45d5e145b85a7aafc42/pkgs/tools/admin/azure-cli/default.nix#L26-L32

[^1]: https://github.com/Azure/azure-cli/blob/azure-cli-2.48.1/src/azure-cli/setup.py#L116
[^2]: https://github.com/Azure/azure-cli/blob/azure-cli-2.48.1/src/azure-cli/requirements.py3.Linux.txt#L72

**Comment by Janik-Haag** [comment]
@jfroche can you update to the new version like Emily suggested and rebase so we can get this merged?

**Comment by AkechiShiro** [comment]
@jfroche if you're too busy to try and give it a try, I can give a try to fork your work and actually update and rebase on latest commits.

**Comment by SuperSandro2000** [comment]
Wer can do the next updates in a new PR

**Comment by AkechiShiro** [comment]
Thanks @SuperSandro2000 !

**Pull request merged**
